flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SourceCoordinator.java

```diff
@@ -101,16 +115,71 @@ Licensed to the Apache Software Foundation (ASF) under one
 
     public SourceCoordinator(
```

Comment:
nit: Could we drop the ctor, already? I believe there are not many places we would need to change.

Reply:
we could, but do we like better a bit of code duplication with passing `WATERMARK_ALIGNMENT_DISABLED` parameter (and making it "more public" api) or having an extra constructor? I would lean slightly towards easier construction in tests and less duplicated code.

Reply:
I am fine with either.